src/deepsparse/transformers/utils/helpers.py

```diff
@@ -33,11 +33,45 @@
     "override_config",
     "process_generation_config",
     "validate_session_ids",
+    "set_generated_length",
 ]
 
 _LOGGER = logging.getLogger(__name__)
 
 
+def set_generated_length(
+    max_length: int,
+    prompt_tokens_length: int,
+    sequence_length: int,
+    prompt_sequence_length: int,
+    max_new_tokens: Union[int, None],
+):
+    """
+    Determine the length of the generated tokens. The maxmimum number of tokens
+    is the sequence_length - prompt_sequence_length. If this value is smaller than the
+    max_length or max_new_tokens provided input by the user, use this value to cap
+    the number of tokens generated. Otherwise, use max_length or max_new_tokens
+    if provided.
+
+    :param max_length: max_length attribtue, provided as input during inference
+    :param prompt_tokens_length: the number of prompt tokens used as part of the
+    generated output
```

Comment:
nit: we generally indent continuation lines; this was initially adopted cause we used Sphinx to generate docs. But not a requirement anymore!